### PR TITLE
model: add schema name, table name to ddl job.

### PR DESCRIPTION
**Pull request by crazycs520**
<!--
Thank you for contributing to TiDB! Please read TiDB's [CONTRIBUTING](https://github.com/pingcap/tidb/blob/master/CONTRIBUTING.md) document **BEFORE** filing this PR.
-->

### What problem does this PR solve? <!--add issue link with summary if exists-->
* add schema name, table name, end_ts to ddl job.

Related parser PR: https://github.com/pingcap/parser/pull/422

Before:
The schema name and table name maybe "" after the database or table droped.
```sql
mysql>admin show ddl jobs
+--------+---------+------------+---------------+--------------+-----------+----------+-----------+-----------------------------------+--------+
| JOB_ID | DB_NAME | TABLE_NAME | JOB_TYPE      | SCHEMA_STATE | SCHEMA_ID | TABLE_ID | ROW_COUNT | START_TIME                        | STATE  |
+--------+---------+------------+---------------+--------------+-----------+----------+-----------+-----------------------------------+--------+
| 105    |         |            | drop schema   | none         | 100       | 0        | 0         | 2019-08-01 14:01:31.355 +0800 CST | synced |
| 104    |         |            | drop table    | none         | 100       | 102      | 0         | 2019-08-01 14:00:40.406 +0800 CST | synced |
| 103    |         |            | create table  | public       | 100       | 102      | 0         | 2019-08-01 14:00:37.955 +0800 CST | synced |
| 101    |         |            | create schema | public       | 100       | 0        | 0         | 2019-08-01 14:00:32.953 +0800 CST | synced |
| 99     |         |            | drop schema   | none         | 95        | 0        | 0         | 2019-08-01 14:00:23.752 +0800 CST | synced |
| 98     |         |            | create table  | public       | 95        | 97       | 0         | 2019-08-01 14:00:12.502 +0800 CST | synced |
| 96     |         |            | create schema | public       | 95        | 0        | 0         | 2019-08-01 14:00:00.304 +0800 CST | synced |
| 94     |         |            | drop schema   | none         | 90        | 0        | 0         | 2019-08-01 13:59:57.807 +0800 CST | synced |
| 93     |         |            | create table  | public       | 90        | 92       | 0         | 2019-08-01 13:54:00.504 +0800 CST | synced |
| 91     |         |            | create schema | public       | 90        | 0        | 0         | 2019-08-01 13:53:53.553 +0800 CST | synced |
+--------+---------+------------+---------------+--------------+-----------+----------+-----------+-----------------------------------+--------+
```
This PR
(DDL 93 and 91 don't have name because it is the old version ddl job.)
```sql
mysql>admin show ddl jobs                                                                                                                                                                                                                                                                                       │| synced |
+--------+---------+------------+---------------+--------------+-----------+----------+-----------+-----------------------------------+-----------------------------------+--------+                                                                                                                            │| 104    |         |            | drop table    | none         | 10
| JOB_ID | DB_NAME | TABLE_NAME | JOB_TYPE      | SCHEMA_STATE | SCHEMA_ID | TABLE_ID | ROW_COUNT | START_TIME                        | END_TIME                          | STATE  |                                                                                                                            │0       | 102      | 0         | 2019-08-01 14:00:40.406 +0800 CST
+--------+---------+------------+---------------+--------------+-----------+----------+-----------+-----------------------------------+-----------------------------------+--------+                                                                                                                            │| synced |
| 105    | test    |            | drop schema   | none         | 100       | 0        | 0         | 2019-08-01 14:01:31.355 +0800 CST | 2019-08-01 14:01:31.556 +0800 CST | synced |                                                                                                                            │| 103    |         |            | create table  | public       | 10
| 104    | test    | t1         | drop table    | none         | 100       | 102      | 0         | 2019-08-01 14:00:40.406 +0800 CST | 2019-08-01 14:00:40.606 +0800 CST | synced |                                                                                                                            │0       | 102      | 0         | 2019-08-01 14:00:37.955 +0800 CST
| 103    | test    | t1         | create table  | public       | 100       | 102      | 0         | 2019-08-01 14:00:37.955 +0800 CST | 2019-08-01 14:00:38.005 +0800 CST | synced |                                                                                                                            │| synced |
| 101    | test    |            | create schema | public       | 100       | 0        | 0         | 2019-08-01 14:00:32.953 +0800 CST | 2019-08-01 14:00:33.004 +0800 CST | synced |                                                                                                                            │| 101    |         |            | create schema | public       | 10
| 99     | test    |            | drop schema   | none         | 95        | 0        | 0         | 2019-08-01 14:00:23.752 +0800 CST | 2019-08-01 14:00:23.952 +0800 CST | synced |                                                                                                                            │0       | 0        | 0         | 2019-08-01 14:00:32.953 +0800 CST
| 98     | test    | t1         | create table  | public       | 95        | 97       | 0         | 2019-08-01 14:00:12.502 +0800 CST | 2019-08-01 14:00:12.557 +0800 CST | synced |                                                                                                                            │| synced |
| 96     | test    |            | create schema | public       | 95        | 0        | 0         | 2019-08-01 14:00:00.304 +0800 CST | 2019-08-01 14:00:00.356 +0800 CST | synced |                                                                                                                            │| 99     |         |            | drop schema   | none         | 95
| 94     | test    |            | drop schema   | none         | 90        | 0        | 0         | 2019-08-01 13:59:57.807 +0800 CST | 2019-08-01 13:59:57.952 +0800 CST | synced |                                                                                                                            │        | 0        | 0         | 2019-08-01 14:00:23.752 +0800 CST
| 93     |         |            | create table  | public       | 90        | 92       | 0         | 2019-08-01 13:54:00.504 +0800 CST | 2019-08-01 13:54:00.554 +0800 CST | synced |                                                                                                                            │| synced |
| 91     |         |            | create schema | public       | 90        | 0        | 0         | 2019-08-01 13:53:53.553 +0800 CST | 2019-08-01 13:53:53.601 +0800 CST | synced |                                                                                                                            │| 98     |         |            | create table  | public       | 95
+--------+---------+------------+---------------+--------------+-----------+----------+-----------+-----------------------------------+-----------------------------------+--------+
```

TiDB used to only store the schema ID and table ID in DDL job info, When executing `admin show ddl jobs`, TiDB use `infoschema` to find schema/table name by the schema ID and table ID. But after we drop the database or table, this does not work any more. 

### What is changed and how it works?
* add schema name, table name to the DDL job info.

### Check List <!--REMOVE the items that are not applicable-->

Tests <!-- At least one of them must be included. -->

 - Unit test
Code changes

 - Has exported function/method change

Side effects


Related changes

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -29,6 +29,10 @@ github.com/coreos/go-systemd v0.0.0-20181031085051-9002847aa142/go.mod h1:F5haX7
 github.com/coreos/pkg v0.0.0-20160727233714-3ac0863d7acf/go.mod h1:E3G3o1h8I7cfcXa63jLwjI0eiQQMgzzUDFVpN/nH/eA=
 github.com/coreos/pkg v0.0.0-20180928190104-399ea9e2e55f h1:lBNOc5arjvs8E5mO2tbpBpLoyyu8B6e44T7hJy6potg=
 github.com/coreos/pkg v0.0.0-20180928190104-399ea9e2e55f/go.mod h1:E3G3o1h8I7cfcXa63jLwjI0eiQQMgzzUDFVpN/nH/eA=
+github.com/crazycs520/parser v0.0.0-20190801014535-8fddad2565dd h1:sELKj3l9keUcFBZGKsKC7vK9W+zofZiIDTSlDvHpSc0=
+github.com/crazycs520/parser v0.0.0-20190801014535-8fddad2565dd/go.mod h1:xLjI+gnWYexq011WPMEvCNS8rFM9qe1vdojIEzSKPuc=
+github.com/crazycs520/parser v0.0.0-20190801015123-1039a56acb0c h1:OCvqcYqI/9Aqula24IjLMHGcsDL/xOhsU3Mh3gXBS3A=
+github.com/crazycs520/parser v0.0.0-20190801015123-1039a56acb0c/go.mod h1:xLjI+gnWYexq011WPMEvCNS8rFM9qe1vdojIEzSKPuc=
```
Comment (tangenta) [inline go.sum]: ?

Comment (crazycs520) [reply go.sum]: Need merge parser first, then I will update the go.mod.

File: executor/executor.go [hunk 2]
```diff
@@ -414,15 +414,33 @@ func (e *ShowDDLJobsExec) Next(ctx context.Context, req *chunk.Chunk) error {
...
+			if e.jobs[i].BinlogInfo.TableInfo != nil {
+				tableName = e.jobs[i].BinlogInfo.TableInfo.Name.L
+			}
+			if len(schemaName) == 0 && e.jobs[i].BinlogInfo.DBInfo != nil {
+				schemaName = e.jobs[i].BinlogInfo.DBInfo.Name.L
+			}
+		}
+		// For compatibility, the old version of DDL Job wasn't store the schema name and table name.
+		if len(schemaName) == 0 {
+			schemaName = getSchemaName(e.is, e.jobs[i].SchemaID)
```
Comment (AilinKid) [inline executor/executor.go]: Do we have some test case for compatibility? (avoiding code coverage decrease)

Comment (crazycs520) [reply executor/executor.go]: done. PTAL.

File: ddl/ddl_api.go [hunk 3]
```diff
@@ -1183,6 +1183,7 @@ func (d *ddl) CreateTableWithLike(ctx sessionctx.Context, ident, referIdent ast.
 	job := &model.Job{
```
Comment (lonng) [inline ddl/ddl_api.go]: Have all cases added `SchemaName`?
There are 24 modification but got 30 by`grep "&model.Job{" ddl/ddl_api.go | wc -l`.

Comment (crazycs520) [reply ddl/ddl_api.go]: done.

File: executor/executor_test.go [hunk 4]
```diff
@@ -417,6 +424,43 @@ func (s *testSuiteP1) TestAdmin(c *C) {
...
+	job.SchemaName = ""
+	err = t.AddHistoryDDLJob(job)
+	c.Assert(err, IsNil)
+	err = tk.Se.StmtCommit()
+	c.Assert(err, IsNil)
+	err = tk.Se.CommitTxn(context.Background())
+	c.Assert(err, IsNil)
+
+	//err = txn.Commit(context.Background())
+	//c.Assert(err, IsNil)
```
[the remark this answers is not in the document]
Comment (crazycs520) [reply executor/executor_test.go]: done.

**Pull request merged**